waspc/data/Generator/templates/react-app/src/auth/forms/Auth.tsx

```diff
@@ -42,590 +29,67 @@ const HeaderText = styled('h2', {
-          {=# isEmailAuthEnabled =}
-          <FormItemGroup>
-            <FormLabel>E-mail</FormLabel>
-            <FormInput
-              type="email"
-              required
-              value={emailFieldVal}
-              onChange={e => setEmailFieldVal(e.target.value)}
-              disabled={isLoading}
-            />
-          </FormItemGroup>
-          {=/ isEmailAuthEnabled =}
-          <FormItemGroup>
-            <FormLabel>Password</FormLabel>
-            <FormInput
-              type="password"
-              required
-              value={passwordFieldVal}
-              onChange={e => setPasswordFieldVal(e.target.value)}
-              disabled={isLoading}
-            />
-          </FormItemGroup>
-
-          <FormItemGroup>
-            <SubmitButton type="submit" disabled={isLoading}>{cta}</SubmitButton>
-          </FormItemGroup>
-        </UserPassForm>
-      {=/ isAnyPasswordBasedAuthEnabled =}
-  </>)
-
   return (
     <Container className={customTheme}>
       <div>
-        {logo && (
-          <img style={logoStyle} src={logo} alt='Your Company' />
-        )}
+        {logo && (<img style={logoStyle} src={logo} alt='Your Company' />)}
         <HeaderText>{title}</HeaderText>
       </div>
 
-      {errorMessage && (<ErrorMessage>
-        {errorMessage.title}{errorMessage.description && ': '}{errorMessage.description}
-      </ErrorMessage>)}
-      {successMessage && <SuccessMessage>{successMessage}</SuccessMessage>}
-      {(state === 'login' || state === 'signup') && loginSignupForm}
-      {=# isEmailAuthEnabled =}
-      {state === 'forgot-password' && (<ForgotPasswordForm
-        isLoading={isLoading}
-        setIsLoading={setIsLoading}
-        setErrorMessage={setErrorMessage}
-        setSuccessMessage={setSuccessMessage}
-      />)}
-      {state === 'reset-password' && (<ResetPasswordForm
-        isLoading={isLoading}
-        setIsLoading={setIsLoading}
-        setErrorMessage={setErrorMessage}
-        setSuccessMessage={setSuccessMessage}
-      />)}
-      {state === 'verify-email' && (<VerifyEmailForm
-        isLoading={isLoading}
-        setIsLoading={setIsLoading}
-        setErrorMessage={setErrorMessage}
-        setSuccessMessage={setSuccessMessage}
-      />)}
-      {=/ isEmailAuthEnabled =}
+      {errorMessage && (
+        <MessageError>
+          {errorMessage.title}{errorMessage.description && ': '}{errorMessage.description}
+        </MessageError>
+      )}
+      {successMessage && <MessageSuccess>{successMessage}</MessageSuccess>}
+      <AuthContext.Provider value={{ isLoading, setIsLoading, setErrorMessage, setSuccessMessage }}>
+        {(state === 'login' || state === 'signup') && (
+          <LoginSignupForm
+            state={state}
+            socialButtonsDirection={socialButtonsDirection}
+          />
+        )}
+        {=# isEmailAuthEnabled =}
+        {state === 'forgot-password' && (<ForgotPasswordForm />)}
```

Comment:
is this some TS thing when you are using string but it's actually a type (State)? So you can't put any string as a state, compiler would complain?

Reply:
Yes sir 👍